azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/AzureKeyVaultCredentialViaCertificate.java

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.keyvault.spring;
+
+import com.microsoft.aad.adal4j.AsymmetricKeyCredential;
+import com.microsoft.aad.adal4j.AuthenticationContext;
+import com.microsoft.aad.adal4j.AuthenticationResult;
+import com.microsoft.azure.keyvault.authentication.KeyVaultCredentials;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.security.*;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.Enumeration;
+import java.util.concurrent.*;
+
+/**
+ * Created by umar on 23/2/18.
+ */
+public class AzureKeyVaultCredentialViaCertificate extends KeyVaultCredentials {
```

Comment:
How about naming it `KeyVaultCertificateCredentials`.
- It is already in an Azure package. So `Azure` is kind of redundant.